### ISSUE TITLE
Building libmodplug fails

### ISSUE THREAD

**Issue by rboy1** [rdp/ffmpeg-windows-build-helpers#275]
When trying to build the latest build of ffmpeg it fails with this error:

> making /ffmpeg/sandbox/win32/libmodplug_git as $ PATH=/ffmpeg/sandbox/cross_compilers/mingw-w64-i686/bin:$PATH make  -j 4
> 
> CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /ffmpeg/sandbox/win32/libmodplug_git/missing aclocal-1.15 -I m4
> CDPATH="${ZSH_VERSION+.}:" && cd . && /bin/bash /ffmpeg/sandbox/win32/libmodplug_git/missing autoconf
>  cd . && /bin/bash /ffmpeg/sandbox/win32/libmodplug_git/missing automake-1.15 --gnu
> configure.ac:21: error: required file './compile' not found
> configure.ac:21:   'automake --add-missing' can install 'compile'

I believe the solution is to run automake with the --add-missing parameter, do you think this can be patched up through the build script?

**Comment by rboy1** [comment]
@rdp here's the solution

modify the method from:

```
build_libmodplug() {
  do_git_checkout https://github.com/Konstanty/libmodplug.git
  cd libmodplug_git
    sed -i.bak 's/__declspec(dllexport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h" #strip DLL import/export directives
    sed -i.bak 's/__declspec(dllimport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h"
    generic_configure_make_install # or could use cmake I guess
  cd ..
}
```

to this:

```
build_libmodplug() {
  do_git_checkout https://github.com/Konstanty/libmodplug.git
  cd libmodplug_git
    sed -i.bak 's/__declspec(dllexport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h" #strip DLL import/export directives
    sed -i.bak 's/__declspec(dllimport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h"
    if [[ ! -f "configure" ]]; then
      autoreconf -fiv || exit 1
      automake --add-missing || exit 1
    fi
    generic_configure_make_install # or could use cmake I guess
  cd ..
}
```

**Comment by hydra3333** [comment]
Just checking in ... not something like this (untested) ?
```
build_libmodplug() {
echo "---------------------------------------------------------------------------------------------------"
echo "---------------------------------------------------------------------------------------------------"
echo "build_libmodplug \"${1}\""
echo "---------------------------------------------------------------------------------------------------"
echo "---------------------------------------------------------------------------------------------------"
  do_git_checkout https://github.com/Konstanty/libmodplug.git
  ## was
  #cd libmodplug_git
  #  sed -i.bak 's/__declspec(dllexport)//' "./src/modplug.h" || exit 1 # ignore failures  #strip DLL import/export directives
  #  generic_configure_make_install # or could use cmake I guess
  #  sed -i.bak 's/-lmodplug.*/-lmodplug -lstdc++/\' "$PKG_CONFIG_PATH/libmodplug.pc"  # ignore failures
  #  sed -i.bak 's/__declspec(dllexport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h" # ignore failures  #strip DLL import/export directives
  #cd ..
  ## is
  cd libmodplug_git
    sed -i.bak 's/__declspec(dllexport)//' "./src/modplug.h" || exit 1 # ignore failures  #strip DLL import/export directives
    sed -i.bak 's/__declspec(dllexport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h" #strip DLL import/export directives
    if [[ ! -f "configure" ]]; then
      autoreconf -fiv || exit 1
      automake --add-missing || exit 1
    fi
    generic_configure_make_install # or could use cmake I guess
    sed -i.bak 's/-lmodplug.*/-lmodplug -lstdc++/\' "$PKG_CONFIG_PATH/libmodplug.pc"  # ignore failures
    sed -i.bak 's/__declspec(dllexport)//' "$mingw_w64_x86_64_prefix/include/libmodplug/modplug.h" # ignore failures  #strip DLL import/export directives
  cd ..
}

```

**Comment by rboy1** [comment]
The earlier one worked for me. I delete the downloaded repository and rebuilt it. I didn't try to rebuild it after a successful build.

**Comment by rboy1** [comment]
The patch I posted above worked perfectly on a rebuild also. Rebuilt from scratch and incrementally with the patch and it's working now

@rdp

**Comment by rdp** [comment]
OK hopefully fixed in  0a6bc96 thanks again!

**Comment by rdp** [comment]
FWIW had this happen again recently.  Fix was to nuke the modplug_git folder and just reclone :)
